pennylane_lightning/core/src/utils/Util.hpp

```diff
@@ -574,4 +574,22 @@ std::vector<T1> cast_vector(const std::vector<T0> &vec) {
     return result;
 }
 
+/**
+ * @brief Check if two vectors are disjoint.
+ * @tparam T Data type.
+ * @param v1 First vector.
+ * @param v2 Second vector.
+ *
+ * @return bool True if the vectors are disjoint, false otherwise.
+ */
+template <typename T = std::size_t>
+bool areVecsDisjoint(const std::vector<T> &v1, const std::vector<T> &v2) {
+    std::set<T> s0(v1.begin(), v1.end());
+    for (const auto &element : v2) {
+        if (s0.find(element) != s0.end()) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
```suggestion
    std::set<int> s0(v1.begin(), v1.end());
    return std::none_of(v2.begin(), v2.end(), [&](int element) {
        return s0.find(element) != s0.end();
    });
```

Reply:
Could you please elaborate more about the suggestions? Would it lead performance regession?

Reply:
Do not worry. If we have a regression in the performance it is better to keep the current implementation. I thought that using std algorithm would achieve some performance. Thank you for corroborating the suggestion.  :rocket: